test/performance/specs/front-end-block-theme.spec.js

```diff
@@ -19,19 +14,30 @@ test.describe( 'Front End Performance', () => {
 		await requestUtils.activateTheme( 'twentytwentythree' );
 	} );
 
-	test.afterAll( async ( { requestUtils } ) => {
-		saveResultsFile( __filename, results );
+	test.afterAll( async ( { requestUtils }, testInfo ) => {
+		await testInfo.attach( 'results', {
+			body: JSON.stringify( results, null, 2 ),
+			contentType: 'application/json',
+		} );
 		await requestUtils.activateTheme( 'twentytwentyone' );
 	} );
 
-	test( 'Report TTFB, LCP, and LCP-TTFB', async ( { page } ) => {
-		let i = 16;
-		while ( i-- ) {
+	test( 'Report TTFB, LCP, and LCP-TTFB', async ( { browser } ) => {
+		const samples = 16;
+		const throwaway = 0;
+		const rounds = samples + throwaway;
+
+		for ( let i = 0; i < rounds; i++ ) {
+			const tmpContext = await browser.newContext( {
+				storageState: {}, // User will be logged out.
+			} );
+			const tmpPage = await tmpContext.newPage();
```

Comment:
Interesting nuance to test as a logged out user 👍 

Is it necessary to create the new context within the loop or could that be done only once before? Is that just to ensure empty caches?

Reply:
By always starting from a fresh state, we aim to reduce the number of variables and ensure that each test run is as similar as possible to the others. The variables can be caching, state side-effects, memory management, etc.

Reply:
@swissspidy we did discover that re-using a session, even after telling it to clear cache, was biasing the performance results above the difference caused by the vast majority of performance-impacting changes.

wherever possible in the performance tests, we should create a new page, new tab, new session, to avoid this.